securedrop/journalist_app/admin.py

```diff
@@ -33,9 +35,30 @@ def manage_config():
             f = form.logo.data
             static_filepath = os.path.join(config.SECUREDROP_ROOT,
                                            "static/i/logo.png")
+            temp_static_filepath = os.path.join(config.SECUREDROP_ROOT,
+                                                "static/i/temp_logo.png")
+            with open(static_filepath) as img_file:
+                current_img = img_file.read()
             f.save(static_filepath)
-            flash(gettext("Image updated."), "logo-success")
-            return redirect(url_for("admin.manage_config"))
+
+            try:
+                with Image.open(static_filepath) as im:
+                    imcopy = im.copy()
+                    imcopy.thumbnail((500, 450), resample=3)
+                    imcopy.save(temp_static_filepath, "PNG")
+                if os.path.exists(static_filepath):
+                    os.rename(temp_static_filepath, static_filepath)
+                flash(gettext("Image updated."), "logo-success")
+
+            except IOError:
```

Comment:
I would change this to `except Exception` because I'm fairly sure that `Image` raises Pillow specific errors on all sorts of things, and we want to alert the user in all cases.

Reply:
I'll update it